app/controllers/resource_pool_controller.rb

```diff
@@ -6,66 +6,30 @@ class ResourcePoolController < ApplicationController
 
   include Mixins::GenericListMixin
   include Mixins::GenericSessionMixin
+  include Mixins::GenericShowMixin
 
-  def show
-    @display = params[:display] || "main" unless pagination_or_gtl_request?
-
-    @lastaction = "show"
-    @showtype   = "config"
-    @record = identify_record(params[:id])
-    return if record_no_longer_exists?(@record)
-
-    @gtl_url = "/show"
-    drop_breadcrumb({:name => _("Resource Pools"),
-                     :url  => "/resource_pool/show_list?page=#{@current_page}&refresh=y"}, true)
-
-    case @display
-    when "download_pdf", "main", "summary_only"
-      get_tagdata(@record)
-      txt = @record.vapp ? _("(vApp)") : ""
-      drop_breadcrumb(:name => _("%{name} %{text} (Summary)") % {:name => @record.name, :text => txt},
-                      :url  => "/resource_pool/show/#{@record.id}")
-      @showtype = "main"
-      set_summary_pdf_data if ["download_pdf", "summary_only"].include?(@display)
-
-    when "vms"
-      drop_breadcrumb(:name => _("%{name} (Direct VMs)") % {:name => @record.name},
-                      :url  => "/resource_pool/show/#{@record.id}?display=vms")
-      @view, @pages = get_view(Vm, :parent => @record)  # Get the records (into a view) and the paginator
-      @showtype = "vms"
-
-    when "descendant_vms"
-      drop_breadcrumb(:name => _("%{name} (All VMs - Tree View)") % {:name => @record.name},
-                      :url  => "/resource_pool/show/#{@record.id}?display=descendant_vms&treestate=true")
-      @showtype = "config"
-
-      self.x_active_tree = :datacenter_tree
-      @datacenter_tree = TreeBuilderDatacenter.new(:datacenter_tree, :datacenter, @sb, true, @record)
-
-    when "all_vms"
-      drop_breadcrumb(:name => "%{name} (All VMs)" % {:name => @record.name},
-                      :url  => "/resource_pool/show/#{@record.id}?display=all_vms")
-      @view, @pages = get_view(Vm, :parent => @record, :association => "all_vms") # Get the records (into a view) and the paginator
-      @showtype = "vms"
+  def self.display_methods
+    %w(vms descendant_vms all_vms resource_pools)
+  end
 
-    when "clusters"
-      drop_breadcrumb(:name => _("%{name} (All %{title})") % {:name => @record.name, :title => title_for_clusters},
-                      :url  => "/resource_pool/show/#{@record.id}?display=clusters")
-      @view, @pages = get_view(EmsCluster, :parent => @record)  # Get the records (into a view) and the paginator
-      @showtype = "clusters"
+  def display_descendant_vms
+    @showtype = "config"
+    drop_breadcrumb(:name => _("%{name} (All VMs - Tree View)") % {:name => @record.name},
```

Comment:
I am trying to understand why you crop a breadcrumb here and not in  `display_all_vms`

Seems to be a similar situation to me: the generic breadcumb dropped in `nested_list` is a different one to the one you need.

In one case you resolve this by ignoring it and in the othe case you resolve it by dropping one more. Both solutions seem wrong to me.

I'd say that the solution to the exceptions here is incorrect and you need to figure out a different one.